eng/packaging.targets

```diff
@@ -42,4 +51,63 @@
     <MSBuild Projects="$(PkgProjPath)"
              Targets="Build" />
   </Target>
+    
+  <Target Name="AddRefAssemblies" Condition="'$(ExcludeFromPackage)' == '' and '$(TargetsAnyOS)' == 'true'"> 
+    <PropertyGroup>
+      <RefPath>$(BaseOutputPath)/ref/$(TargetFrameworkWithoutSuffix)-$(Configuration)/$(TargetName).dll</RefPath>
+    </PropertyGroup>
+    
+    <ItemGroup Condition="Exists('$(RefPath)')">
+      <TfmSpecificPackageFile Include="$(RefPath)">
+        <PackagePath>ref/$(TargetFrameworkWithoutSuffix)</PackagePath>
+      </TfmSpecificPackageFile>
+      <TfmSpecificPackageFile Include="$(XmlDocFileRoot)1033\$(TargetName).xml" Condition="Exists('$(XmlDocFileRoot)1033\$(TargetName).xml')">
```

Comment:
Existing problem with the code that exists for lib, but shouldn't we key this off of some state that indicates what our doc source of truth should be?  Haven't some libs switched to using source-generated docs (@safern @carlossanlop)

Reply:
Yes. We should introduce a property that csproj's that have switched to triple slash set to true, and if that property is set to true, then, use the Roslyn generated xml.

Reply:
Can you make sure to file an issue to track that?

Reply:
Sure: https://github.com/dotnet/runtime/issues/54761